src/components/DropzoneArea.js

```diff
@@ -17,117 +10,85 @@ const splitDropzoneAreaProps = (props) => {
-                    const file = await createFileFromUrl(url);
-                    const data = await readFile(file);
-
-                    return {
-                        file,
-                        data,
-                    };
-                })
-            );
 
-            this.setState((state) => ({
-                fileObjects: [].concat(
-                    state.fileObjects,
-                    fileObjs
-                ),
-            }), this.notifyFileChange);
-        } catch (err) {
-            console.log(err);
-        }
-    }
-
-    addFiles = async(newFileObjects) => {
-        const {filesLimit} = this.props;
-        // Update component state
-        this.setState((state) => {
-            // Handle a single file
-            if (filesLimit <= 1) {
-                return {
-                    fileObjects: [].concat(newFileObjects[0]),
-                };
-            }
-
-            // Handle multiple files
-            return {
-                fileObjects: [].concat(
-                    state.fileObjects,
-                    newFileObjects
-                ),
-            };
-        }, this.notifyFileChange);
-    }
-
-    deleteFile = (removedFileObj, removedFileObjIdx) => {
-        event.stopPropagation();
-
-        const {onDelete} = this.props;
-        const {fileObjects} = this.state;
-
-        // Calculate remaining fileObjects array
-        const remainingFileObjs = fileObjects.filter((fileObject, i) => {
-            return i !== removedFileObjIdx;
-        });
-
-        // Notify removed file
-        if (onDelete) {
-            onDelete(removedFileObj.file);
-        }
-
-        // Update local state
-        this.setState({
-            fileObjects: remainingFileObjs,
-        }, this.notifyFileChange);
-    }
-
-    render() {
-        const [, dropzoneAreaProps] = splitDropzoneAreaProps(this.props);
-        const {fileObjects} = this.state;
-
-        return (
-            <DropzoneAreaBase
-                {...dropzoneAreaProps}
-                fileObjects={fileObjects}
-                onAdd={this.addFiles}
-                onDelete={this.deleteFile}
-            />
-        );
-    }
-}
+const DropzoneArea = ({
+    clearOnUnmount,
+    initialFiles,
+    onChange,
+    onDelete,
+    filesLimit,
+    getFileAddedMessage,
+    getFileLimitExceedMessage,
+    getFileRemovedMessage,
+    getDropRejectMessage,
+    onDrop,
+    onDropRejected,
+    acceptedFiles,
+    alertSnackbarProps,
+    disableRejectionFeedback,
+    dropzoneClass,
+    dropzoneParagraphClass,
+    dropzoneProps,
+    dropzoneText,
+    getCols,
+    getPreviewIcon,
+    inputProps,
+    maxFileSize,
+    previewChipProps,
+    previewGridClasses,
+    previewGridProps,
+    previewText,
+    previewType,
+    showAlerts,
+    showFileNames,
+    useChipsForPreview,
+}) => {
+    const {
+        deleteFile,
+        addFiles,
+        fileObjects,
+    } = useDropzoneArea({
```

Comment:
I think this should be named `useDropzoneAreaState` and maybe we can think to expose it so that it can be used by developers.

Also consider changing `deleteFile` and `addFiles` to `onDelete` and `onAdd` to allow something like this

```js
const dropzoneAreaState = useDropzoneAreaState({
...
});

...

return (
  <DropzoneAreaBase
        ....
        ....
        {...dropzoneAreaState}
  />
```

Reply:
@panz3r , I was writing some comments to document the what hookt `useDropzoneArea` does, and I came up with `"Holds the fileObjects state and returns handler methods to remove and add files", this lead me to an idea for a better name like perhaps `useFIles` what do you think about that name? To be honest I'm not very good at naming hooks so whatever makes it easiest to understand is good 😀

Reply:
Also with the handler naming maybe I should use `handle` as prefix rather than `on` since we are providing the function  rather than it being one passed in?

Reply:
> this lead me to an idea for a better name like perhaps useFIles`

`useFiles` definitely makes sense 👍 

> Also with the handler naming maybe I should use handle as prefix rather than on since we are providing the function rather than it being one passed in?

Ok, sounds good 👍

Reply:
Hi @panz3r  I have considered your idea for ` {...dropzoneAreaState}` however I feel that unless we can think of an absolutely brilliant name for this, this may lead to some confusion because it will ultimately be a grouping of whatever state variables ,handlers are returned by the useFiles,, 

The best name I can think of at the moment would be ...fileObjectsAndHandlers

I'm wondering if this may lead to more confusion at any point e.g. (Which props has ...fileObjectsAndHandlers got in it again?) However this is something trivial that can be easily changed in the future